### PR TITLE
#51: check if file exists before send

### PR DESCRIPTION
**Pull request by pmlopes**
Solves: https://github.com/vert-x3/issues/issues/51

Signed-off-by: Paulo Lopes <paulo@mlopes.net>

### CODE REVIEW

File: src/test/java/io/vertx/test/core/HttpTest.java [hunk 1]
```diff
@@ -2295,6 +2295,29 @@ private void sendFile(String fileName, String contentExpected, boolean handler)
   }
 
   @Test
+  public void testSendNonExistingFile() throws Exception {
+    server.requestHandler(req -> {
+      req.response().sendFile("/not/existing/path", event -> {
```
Comment (vietj) [inline src/test/java/io/vertx/test/core/HttpTest.java]: check that this is the correct Context in the callback

**Comment by purplefox** [comment]
I'm not sure why this fix was necessary. In the previous implementation the error is reported fine in the sendFile handler...

**Comment by pmlopes** [comment]
using the included test, without the fix:

```java
req.response().sendFile("/not/existing/path", event -> {
         if (event.failed()) {
           req.response().end("failed");
         }
```

The event state will be `success` and the final response is 200 with content length of 0. With the fix, the event state will be `failed`.

**Comment by purplefox** [comment]
I don't get those results.

If I comment out the code you added in HttpServerResponseImpl and set a breakpoint in the test you added to inspect the result (event) I can see it is failed and has a cause of FileNotFoundException.

**Pull request merged**